modules/resinkernel.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+
+#
+# ** License **
+#
+# Home: http://resin.io
+#
+# Author: Andrei Gherzan <andrei@resin.io>
+#
+
+import unittest
+import os
+import logging
+from .util import *
+from .colorlogging import *
+
+def rmmod(name):
+    ''' Removes a kernel module from current tree '''
+    cmd = "rmmod " + name
+    child = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    out, err = child.communicate()
+    if child.returncode != 0:
+        log.error("resinkernel: rmmod on " + name + " failed.")
+        return False
+    return True
+
+def loaded():
+    ''' Iterate through the currently loaded modules '''
+    cmd = "lsmod"
+    child = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    out, err = child.communicate()
+    if child.returncode != 0:
+        log.error("resinkernel: lsmod failed.")
+        return None
+
+    lsmod = out.decode().split('\n')[1:]
+    for entry in lsmod:
+        entry = entry.strip()
+        if not entry:
+            continue
+        module = entry.split()[0]
+        yield module
+
+def modinfo(name, attr):
+    ''' Returns the modinfo value of the attribute requested '''
+    cmd = "modinfo " + name
+    child = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    out, err = child.communicate()
+    if child.returncode != 0:
+        log.error("resinkernel: No value for " + attr + " in modinfo on " + name + " .")
+        return None
+
+    modinfo = out.decode().split('\n')
+    for entry in modinfo:
+        attribute = entry.split(':', 1)[0].strip()
+        value = entry.split(':', 1)[1].strip()
+        if attribute == attr:
+            return value
+
+    return None
+
+class ResinKernel(object):
+    def CustomLoadedModules(self):
```

Comment:
I use CamelCase for class names as it is recommended in PEP8. Now I do agree that they don't say if the name should start or not with a capital letter but CamelCase can be either). My preference was to start with capital letters for class names as I use CamelCase for methods too (which I do agree that it shouldn't be).